tap_bigquery/sync_bigquery.py

```diff
@@ -207,24 +207,31 @@ def do_sync(config, state, stream):
                             day=row[key].day)
                     elif type(row[key]) == datetime.datetime:
                         r = row[key]
+                    elif row[key] is None:
+                        if prop.type[0] != "null":
+                            raise ValueError("NULL value not allowed by the schema")
+                        r = None
                     else:
                         raise ValueError(
-                            "Record does not match datetime schema %s" %
-                            row[key])
-                    record[key] = r.isoformat()
+                            "Record does not match datetime or None type",
+                            "Row = {}, key = {}".format(row, key)
+                        )
+                    record[key] = r.isoformat() if r else None
                 elif prop.type[1] == "string":
                     record[key] = str(row[key])
                 elif prop.type[1] == "number" and row[key]:
                     record[key] = Decimal(row[key])
                 elif prop.type[1] == "integer" and row[key]:
                     record[key] = int(row[key])
+                elif row[key] is None and prop.type[0] != "null":
```

Comment:
1. This null check, which is applicable to all types (date-time, string, number, and integer) should be moved up front as an independent if-clause at Ln 199.
2. Then `and row[key]` checks won't be necessary at Lines 222 and 224.
3. Also, check at 211 would be redundant after (1).

...let me know if it's becoming too much work and you want me to take over this fix :)

Reply:
Done!